pkg/identity/identity_test.go

```diff
@@ -51,10 +52,6 @@ func (s *IdentityTestSuite) TestReservedID(c *C) {
 	c.Assert(i, Equals, NumericIdentity(3))
 	c.Assert(i.String(), Equals, "unmanaged")
 
-	i = GetReservedID("kube-apiserver")
-	c.Assert(i, Equals, NumericIdentity(7))
-	c.Assert(i.String(), Equals, "kube-apiserver")
-
```

Comment:
This deletion seems surprising. Is this PR removing the reserved identity for "kube-apiserver"? What if kube-apiserver is not a cluster node?

Reply:
When this behavior is enabled, `kube-apiserver` is just a label. The special identity is no longer needed.

I'll add this test logic back, behind the correct flag.

Reply:
Removing this was an oversight anyways; fixed.